React/Views/RCTModalHostView.m

```diff
@@ -89,6 +101,13 @@ - (void)insertReactSubview:(UIView *)subview atIndex:(NSInteger)atIndex
   RCTAssert(_reactSubview == nil, @"Modal view can only have one subview");
   [super insertReactSubview:subview atIndex:atIndex];
   [_touchHandler attachToView:subview];
+#if TARGET_OS_TV
+  if (_onRequestClose) {
```

Comment:
Why we don't want to? Just because of performance reasons or because it changes default dehaviour?

I think there is no logical connections between adding subviews and setting `onRequest` property. Let's imagine that we apply this property inside `render` function only if some condition is true, so it can happene after the view three is established. Does this make sense?

So, I believe we have to override setter `setOnRequsetClose` and establish gesture recognizer there. And we also have to reinfoce it in `insertReactSubview:`.